modules/util/util.js

```diff
@@ -218,6 +218,19 @@ export function utilDisplayName(entity) {
         name = t('inspector.display_name.' + keyComponents.join('_'), tags);
     }
 
+    // if there's still no name found, try addr:housename
+    if (!name && entity.tags['addr:housename']) {
+        name = entity.tags['addr:housename'];
+    }
+
+    // as a last resort, use the street address as a name
+    if (!name && entity.tags['addr:housenumber'] && entity.tags['addr:street']) {
```

Comment:
`addr:unit` may also be necessary to differentiate address points.

Reply:
It seems to me that you'd want to display the housenumber if there was no name and addr:street tag

Reply:
@1ec5 I just added `addr:unit`, but I also just read https://github.com/gravitystorm/openstreetmap-carto/issues/3568... In New Zealand it is `{unit}/{housenumber}` or `{unit}, {housenumber}`

@pnorman I didn't want to cause confusion by having a random number showing on the map if there is `addr:housenumber` but no `addr:street`. But if you think that's a good idea, I'll make that change

Reply:
> In New Zealand it is {unit}/{housenumber} or {unit}, {housenumber}

Yeah, ultimately it’ll be more effective to read the existing address_formats.json, which defines the order of the fields in the address field after geocoding, and join them with spaces. That would require a bit of refactoring, though, so it probably could be a separate task. For now, I suppose we could leave it up to translators to come up with a reasonable string that may or may not match the local addressing convention but would still be recognizable.